common/tools/dev-tool/src/config/rollup.base.config.ts

```diff
@@ -83,14 +51,14 @@ function ignoreNiseSinonEval(warning: RollupWarning): boolean {
 function ignoreChaiCircularDependency(warning: RollupWarning): boolean {
   return (
     warning.code === "CIRCULAR_DEPENDENCY" &&
-    matchesPathSegments(warning.importer, ["node_modules", "chai"])
+    matchesPathSegments(warning.ids?.[0], ["node_modules", "chai"])
   );
 }
 
 function ignoreRheaPromiseCircularDependency(warning: RollupWarning): boolean {
   return (
     warning.code === "CIRCULAR_DEPENDENCY" &&
-    matchesPathSegments(warning.importer, ["node_modules", "rhea-promise"])
+    matchesPathSegments(warning.ids?.[0], ["node_modules", "rhea-promise"])
```

Comment:
can you verify by with event-hubs or service-bus? When I tried, both id, or ids are undefined for rhea-promise warnings. I guess that was why we used `importer`.  Even though Rollup v3 changed the type of RollupWarning, it looks that those warnings still have the same shape in JavaScript.  I had to use `message` 

Maybe also worth to log a rollup/rollup plugin bug if we expect valid ids.

Reply:
They build fine:
```
$ rushx build
Your version of Node.js (19.9.0) has not been tested with this release of Rush. Please consider installing a newer version of the "@microsoft/rush" package, or downgrading Node.js.

Found configuration in /home/codespace/window1/rush.json

Rush Multi-Project Build Tool 5.106.0 - Node.js 19.9.0 (unstable)
Your version of Node.js (19.9.0) is an odd-numbered release. These releases frequently have bugs. Please consider installing a Long Term Support (LTS) version instead.

> "npm run clean && tsc -p . && dev-tool run bundle && api-extractor run --local && npm run build:types"


> @azure/event-hubs@5.11.3 clean
> rimraf dist dist-* types *.tgz *.log

[bundle] Created production CommonJS bundle.
[bundle] Created browser testing bundle.

api-extractor 7.36.4  - https://api-extractor.com/

Using configuration from ./api-extractor.json
Analysis will use the bundled TypeScript version 5.0.4

API Extractor completed successfully

> @azure/event-hubs@5.11.3 build:types
> downlevel-dts types/latest types/3.1

```

Reply:
oh right, event-hubs and service-bus have rollup v2 because they have customer test config. `build:test` uses a different rollup.test.config.js which may also needs.  could be another PR to upgrade them.

Reply:
Ah the libraries that polyfill node builtins are still on v2. Addressing them in a different PR sounds good.